### ISSUE TITLE
[BUG] v3.1.0 Cargo build failure: "Unresolved import `watchexec::ignore`"

### ISSUE THREAD

**Issue by monoflo** [oknozor/toml-bombadil#194]
**Describe the bug**
Attempting to `cargo install` the crate results in an `error[E0432]`:

```
$ cargo install toml-bombadil
...
   Compiling watchexec v2.0.2
   Compiling git2 v0.14.4
   Compiling toml-bombadil v3.1.0
error[E0432]: unresolved import `watchexec::ignore`
   --> /home/rlagasse/.cargo/registry/src/github.com-1ecc6299db9ec823/toml-bombadil-3.1.0/src/lib.rs:224:13
    |
224 |             ignore,
    |             ^^^^^^ no `ignore` in the root

For more information about this error, try `rustc --explain E0432`.
error: could not compile `toml-bombadil` due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `toml-bombadil v3.1.0`, intermediate artifacts can be found at `/tmp/cargo-installSLdD1r`

Caused by:
  build failed
```

**To Reproduce**

Run `cargo install toml-bombadil --version 3.1.0`

**Additional context**

It seems as though there is no `watchexec::ignore` to be `use`d in `watchexec v2.0.2`:

https://docs.rs/watchexec/2.0.2/watchexec/index.html?search=ignore

**Comment by monoflo** [comment]
I have not experienced the same behavior with `cargo install toml-bombadil --version 3.0.0`. The build succeeds in this version.

I notice that [`v3.0.0` does not depend on `watchexec`](https://github.com/oknozor/toml-bombadil/blob/3.0.0/Cargo.toml#L16), but [`v3.1.0` does.](https://github.com/oknozor/toml-bombadil/blob/3.1.0/Cargo.toml#L30).

In-fact, I am curious that `cargo install` is resolving the dependency version `2.2.0` for `watchexec` instead of the version `2.0.0-pre.14` specified in the `Cargo.toml`.

**Comment by monoflo** [comment]
This build error doesn't occur if I `cargo build` commit `9515982805e9db105617da456cf14f57925d33d8` (tag `3.1.0`):

```
$ cargo build
   Compiling watchexec v2.0.0-pre.14
   Compiling git2 v0.14.3
   Compiling toml-bombadil v3.1.0 (<path>)
    Finished dev [unoptimized + debuginfo] target(s) in 56.79s
$ git describe --always --dirty
9515982
```

**Comment by monoflo** [comment]
Reproduced on the same commit with: `cargo install --path .`
https://github.com/oknozor/toml-bombadil/issues/194#issuecomment-1254108504

There must be some discrepancy between how `cargo` resolves dependencies between `cargo build` and `cargo install`, and I imagine it did *not* like the `-pre.14` suffix for `2.0.0-pre.14`.

**Comment by oknozor** [comment]
Should be fixed when merging #156

**Comment by oknozor** [comment]
Fixed in the upcoming version, meanwhile you can test using `cargo install --git  https://github.com/oknozor/toml-bombadil`.
